streams/src/test/java/org/apache/kafka/streams/state/internals/OffsetCheckpointTest.java

```diff
@@ -56,4 +57,20 @@ public void testReadWrite() throws IOException {
             checkpoint.delete();
         }
     }
+
+    @Test
+    public void testNotWriteEmptyMap() throws IOException {
+        // we do not need to worry about file name unqiueness since this file should not be created
+        File f = new File(TestUtils.tempDirectory().getAbsolutePath(), "kafka.tmp");
+        OffsetCheckpoint checkpoint = new OffsetCheckpoint(f);
+
+        checkpoint.write(Collections.<TopicPartition, Long>emptyMap());
+
+        assertFalse(f.exists());
+
+        assertEquals(Collections.<TopicPartition, Long>emptyMap(), checkpoint.read());
+
+        checkpoint.delete();
```

Comment:
Why do we need a delete here if the file was never created?

Reply:
+1 - this line and the next shouldn't be needed

Reply:
I added this line intentionally to make sure `delete()` would not bark if the file does not exist, i.e. internally it calls `deleteIfExists`.

Reply:
fair enough - we still don't need the assertion though as it is already true as of line 69

Reply:
I'll remove the assertion.